### PR TITLE
Use setError for track error handling

### PR DESCRIPTION
**Pull request by pkerpedjiev**
## Description

> What was changed in this pull request?

Make tracks use `setError` for setting error texts. This lets one track set multiple errors by using the `source` parameter.

> Why is it necessary?

So that tracks can set and display multiple errors. If this functionality wasn't there, the different overlaps would overlap and be illegible.

Fixes #___

## Checklist

- [ ] Set proper GitHub labels (e.g. v1.6+, ignore if you don't know)
- [ ] Unit tests added or updated
- [ ] Documentation added or updated
- [ ] Example(s) added or updated
- [ ] Update schema.json if there are changes to the viewconf JSON structure format
- [ ] Screenshot for visual changes (e.g. new tracks or UI changes)
- [ ] Updated CHANGELOG.md

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -1,8 +1,8 @@
 <!DOCTYPE html>
 <html>
   <head>
-    <meta charset="utf-8">
-    <meta name="viewport" content="width=device-width,initial-scale=1">
+    <meta charset="utf-8" />
```
Comment (manzt) [inline index.html:4]: I'm guessing these changes came from some auto formatting in the editor. I can't see to make out any actual changes to this file, so maybe we want to revert these changes.

File: app/scripts/TiledPixiTrack.js [hunk 2]
```diff
@@ -224,12 +224,6 @@ class TiledPixiTrack extends PixiTrack {
     });
   }
 
-  setError(error) {
-    this.errorTextText = error;
-    this.draw();
-    this.animate();
```
Comment (manzt) [inline app/scripts/TiledPixiTrack.js:230]: Noticing that this `setError` impl that we removed has the side effect of `draw()` + `animate()` while the new base impl just calls `PixiTrack.drawError()`. Are these equivalent?

File: app/scripts/TiledPixiTrack.js [hunk 3]
```diff
@@ -756,18 +773,8 @@ class TiledPixiTrack extends PixiTrack {
...
-          x.tileData && x.tileData.error && `${x.tileId}: ${x.tileData.error}`,
-      )
-      .filter((x) => x);
 
-    if (errors.length) {
-      this.errorTextText = errors.join('\n');
-    } else {
-      this.errorTextText = '';
-    }
+    this.checkForErrors();
```
Comment (manzt) [inline app/scripts/TiledPixiTrack.js:777]: Is this meant to be an inherited method? If not... I'd maybe recommend underscoring it (e.g., `_checkForErrors()`) to indicate an "internal" method.

File: app/scripts/PixiTrack.js [hunk 4]
```diff
@@ -294,15 +296,43 @@ class PixiTrack extends Track {
...
+
+    this.drawError();
+  }
+
   drawError() {
     this.errorText.x = this.position[0] + this.dimensions[0] / 2;
     this.errorText.y = this.position[1] + this.dimensions[1] / 2;
 
-    this.errorText.text = this.errorTextText;
+    let errorTextText = '';
```
Comment (manzt) [inline app/scripts/PixiTrack.js:318]: Small nit, took me a little bit to see what was going on here (filter + reduce). 

Not essential, but maybe we could make that more explicit. E.g.,

```javascript
const errorTextText = Object.values(this.errorTexts)
  .filter(x => x && x.length)
  .reduce((acc, x) => acc ? `${acc}\n${x}` : x, '');
```

**Review by manzt — APPROVED** [review]
Looks great! Some super tiny nits (feel free to ignore).

From the title, I thought something like [`Error.cause`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/cause) might be relevant... but now I see this isn't JavaScript error handling but rather communicating tileset errors.

File: index.html (hunk 1 above)
Comment (pkerpedjiev) [reply index.html:4]: Yeah, I can checkout the original file. These are visual studio auto-formatting changes.

File: app/scripts/TiledPixiTrack.js (hunk 2 above)
Comment (pkerpedjiev) [reply app/scripts/TiledPixiTrack.js:230]: Well the previous draw used to just call `drawError` so we're basically cutting out the middleman.

Comment (pkerpedjiev) [reply app/scripts/TiledPixiTrack.js:230]: The lack of the animate call is slightly more concerning but I've been using this function a fair amount in resglass and haven't noticed any issues so I'm inclined to leave it out.

File: app/scripts/PixiTrack.js (hunk 4 above)
Comment (pkerpedjiev) [reply app/scripts/PixiTrack.js]: You kids and your idiomatic expressions!

**Review by manzt — APPROVED** [review]
LGTM!

**Pull request merged**